packages/manager/apps/pci-savings-plan/src/components/CreatePlanForm/CreatePlanForm.tsx

```diff
@@ -335,7 +337,11 @@ const CreatePlanForm: FC<CreatePlanFormProps> = ({
             <QuantitySelector
               quantity={quantity}
               onMinusClick={() => setQuantity(quantity - 1)}
-              onPlusClick={() => setQuantity(quantity + 1)}
+              onPlusClick={() => {
+                if (quantity <= MAX_QUANTITY) {
+                  setQuantity(quantity + 1);
```

Comment:
To avoid asynchronous issues, we could use `previous` value as parameter:
```suggestion
                  setQuantity(prevQuantity => prevQuantity + 1);
```